--- udftools-1.2/libudffs/unicode.c.orig	2016-06-29 19:13:05.000000000 +0200
+++ udftools-1.2/libudffs/unicode.c	2016-10-27 05:35:27.516758958 +0200
@@ -25,6 +25,7 @@
  * libudffs unicode handling functions
  */
 
+#include <string.h>
 #include "libudffs.h"
 #include "config.h"
 
--- udftools-1.2/mkudffs/file.c.orig	2016-06-29 19:13:05.000000000 +0200
+++ udftools-1.2/mkudffs/file.c	2016-10-27 05:39:23.523422929 +0200
@@ -26,6 +26,7 @@
  */
 
 #include <malloc.h>
+#include <string.h>
 
 #include "libudffs.h"
 #include "file.h"
--- udftools-1.2/cdrwtool/options.c.orig	2016-06-29 19:13:05.000000000 +0200
+++ udftools-1.2/cdrwtool/options.c	2016-10-27 05:45:20.480085516 +0200
@@ -22,6 +22,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <malloc.h>
 
 #include "cdrwtool.h"
