diff -Nur b.udftools-1.0.0b3/Makefile.am n.udftools-1.0.0b3/Makefile.am
--- b.udftools-1.0.0b3/Makefile.am	2002-11-26 08:18:50.000000000 +0100
+++ n.udftools-1.0.0b3/Makefile.am	2006-11-12 00:04:50.000000000 +0100
@@ -1 +1 @@
-SUBDIRS = libudffs mkudffs cdrwtool pktsetup udffsck wrudf doc udfct
+SUBDIRS = libudffs mkudffs cdrwtool pktsetup udffsck wrudf doc udfct include
diff -Nur b.udftools-1.0.0b3/configure.in n.udftools-1.0.0b3/configure.in
--- b.udftools-1.0.0b3/configure.in	2006-11-12 00:04:18.000000000 +0100
+++ n.udftools-1.0.0b3/configure.in	2006-11-12 00:17:11.000000000 +0100
@@ -25,6 +25,7 @@
 AC_SUBST(LTLIBOBJS)
 AC_CHECK_FUNCS([getopt_long],[],[AC_LIBOBJ([getopt])])
 
-AC_CONFIG_FILES(Makefile libudffs/Makefile mkudffs/Makefile cdrwtool/Makefile pktsetup/Makefile udffsck/Makefile wrudf/Makefile doc/Makefile udfct/Makefile)
+AC_CONFIG_FILES(Makefile libudffs/Makefile mkudffs/Makefile cdrwtool/Makefile pktsetup/Makefile udffsck/Makefile wrudf/Makefile doc/Makefile
+				udfct/Makefile include/Makefile)
 
 AC_OUTPUT
diff -Nur b.udftools-1.0.0b3/include/Makefile.am n.udftools-1.0.0b3/include/Makefile.am
--- b.udftools-1.0.0b3/include/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ n.udftools-1.0.0b3/include/Makefile.am	2006-11-12 00:23:40.000000000 +0100
@@ -0,0 +1,10 @@
+pkgincludedir = $(includedir)/udffs
+pkginclude_HEADERS = \
+	 bswap.h \
+	 defaults.h \
+	 ecma_167.h \
+	 getopt.h \
+	 libudffs.h \
+	 osta_udf.h \
+	 udf_endian.h \
+	 udf_lib.h
