--- udftools-1.0.0b3/cdrwtool/cdrwtool.c.old	2007-11-09 13:09:44.000000000 +0100
+++ udftools-1.0.0b3/cdrwtool/cdrwtool.c	2007-11-09 13:10:34.000000000 +0100
@@ -32,6 +32,7 @@
 #include <asm/param.h>
 
 #include <linux/cdrom.h>
+#include <limits.h>
 
 #include "cdrwtool.h"
 #include "../mkudffs/mkudffs.h"
--- udftools-1.0.0b3/pktsetup/pktsetup.c.old	2007-11-09 13:11:28.000000000 +0100
+++ udftools-1.0.0b3/pktsetup/pktsetup.c	2007-11-09 13:11:57.000000000 +0100
@@ -17,6 +17,7 @@
  *   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  *
  */
+#include <limits.h>
 #include <stdio.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
